llm_bench/python/llm_bench_utils/hook_sample_v43.py

```diff
@@ -0,0 +1,235 @@
+        do_sample = generation_config.do_sample
+        if do_sample is True and not isinstance(logits_warper, LogitsProcessorList):
+            raise ValueError(
+                "`do_sample` is set to `True`, `logits_warper` must be a `LogitsProcessorList` instance (it is "
+                f"{logits_warper})."
+            )
+
+        # init attention / hidden states / scores tuples
+        scores = () if (return_dict_in_generate and output_scores) else None
+        raw_logits = () if (return_dict_in_generate and output_logits) else None
+        decoder_attentions = () if (return_dict_in_generate and output_attentions) else None
+        cross_attentions = () if (return_dict_in_generate and output_attentions) else None
+        decoder_hidden_states = () if (return_dict_in_generate and output_hidden_states) else None
+
+        # if model is an encoder-decoder, retrieve encoder attention weights and hidden states
+        if return_dict_in_generate and self.config.is_encoder_decoder:
+            encoder_attentions = model_kwargs["encoder_outputs"].get("attentions") if output_attentions else None
+            encoder_hidden_states = (
+                model_kwargs["encoder_outputs"].get("hidden_states") if output_hidden_states else None
+            )
+
+        # keep track of which sequences are already finished
+        batch_size = input_ids.shape[0]
+        this_peer_finished = False
+        unfinished_sequences = torch.ones(batch_size, dtype=torch.long, device=input_ids.device)
+        model_kwargs = self._get_initial_cache_position(input_ids, model_kwargs)
+
+        while self._has_unfinished_sequences(this_peer_finished, synced_gpus, device=input_ids.device):
+            tic = time.perf_counter()
+            # prepare model inputs
+            model_inputs = self.prepare_inputs_for_generation(input_ids, **model_kwargs)
+
+            # prepare variable output controls (note: some models won't accept all output controls)
+            model_inputs.update({"output_attentions": output_attentions} if output_attentions else {})
+            model_inputs.update({"output_hidden_states": output_hidden_states} if output_hidden_states else {})
+
+            # forward pass to get next token
+            tic_infer = time.perf_counter()
+            outputs = self(**model_inputs, return_dict=True)
+            hook_greedy.tm_infer_list.append(time.perf_counter() - tic_infer)
+
+            if synced_gpus and this_peer_finished:
+                continue  # don't waste resources running the code we don't need
+
+            # Clone is needed to avoid keeping a hanging ref to outputs.logits which may be very large for first iteration
+            # (the clone itself is always small)
+            #next_token_logits = outputs.logits[:, -1, :].clone()
+            
+            # clone causes regression compared with transformers V41.1
```

Comment:
Please explain why clone started to be slow in this case. Does it mean that something goes wrong in another place and we are hiding the effect here?

Reply:
We don't have answer now. Will try to profile it.

Reply:
@peterchen-intel, please confirm that while providing a workaround for the issue in this PR we are not providing a solution for the customers that are using vanilla hf/optimum?

Reply:
From this perspective, we should reproduce the issue and report it to transformers.

Reply:
> ... and report it to transformers
Or to OpenVINO if it turns out that something bad happened to that output tensor a slice of which is being cloned here.

Reply:
@slyalin @eaidova  After the investigation. This .clone() makes the issue CVS-150542, CVS-145996 become worse, it is not the root cause of the regression. Removing it can mitigate the regression (the worse part), but not the fixing for original issue. The original issue is due to the "busy-wait spin" conflict between oneTBB (OV inference) and OMP (torch post processing like greedy search or beam search), the solution is to limit the torch threads for post processing or setting [OMP_WAIT_POLICY|https://gcc.gnu.org/onlinedocs/libgomp/OMP_005fWAIT_005fPOLICY.html]=PASSIVE of OMP to disable busy-wait spin. We will ask oneTBB to support similar feature to disable oneTBB busy-wait spin.